src/utils/device/parseUserAgent.ts

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { getParser, Parser } from 'bowser';
+
+export enum DeviceType {
+    Desktop = 'Desktop',
+    Mobile = 'Mobile',
+    Web = 'Web',
+    Unknown = 'Unknown',
+}
+export type ExtendedDeviceInformation = {
+    deviceType: DeviceType;
+    // eg Google Pixel 6
+    deviceModel?: string;
+    // eg Android 11
+    deviceOperatingSystem?: string;
+    // eg Firefox
+    clientName?: string;
+    // eg 1.1.0
+    clientVersion?: string;
+};
+
+const getDeviceType = (
+    device: Parser.PlatformDetails,
+    browser: Parser.BrowserDetails,
+    appBrand?: string,
+): DeviceType => {
+    // this sucks
+    const safeBrowserName = appBrand && browser.name.indexOf(appBrand) > -1 ? undefined : browser.name;
```

Comment:
:laughing: Is there anything that sucks less? You could use `includes` instead of the `indexOf` check btw..